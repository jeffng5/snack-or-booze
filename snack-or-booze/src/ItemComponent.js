import React from 'react';
import { Redirect, useParams } from 'react-router-dom'
import { Card, CardBody, CardTitle, CardText } from "reactstrap";


function ItemComponent({item, cantFind}) {
    const { id } = useParams() 
    let thing = item.find(thing => thing.id === id);
    if (!item) return <Redirect to={cantFind} />;

return(    
    <section>
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {thing.name}
          </CardTitle>
          <CardText className="font-italic">{thing.description}</CardText>
          <p>
            <b>Recipe:</b> {thing.recipe}
          </p>
          <p>
            <b>Serve:</b> {thing.serve}
          </p>
        </CardBody>
      </Card>
    </section>
  );

}

export default ItemComponent