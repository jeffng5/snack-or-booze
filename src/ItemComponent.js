import React from 'react';
import { Link } from 'react-router-dom'
import { Card, CardBody, CardTitle, CardText } from "reactstrap";



const ItemComponent = ({key, id, description, recipe, serve }) => {




return(    
    <>
    <CardTitle>
      <Card>
        <CardBody>
    {key}
        {id}
        
           
      
          <CardText className="font-italic">{description}
          <p>
            <b>Recipe:</b> {recipe}
          </p>
          <p>
            <b>Serve:</b> {serve}
          </p>
          </CardText>
        </CardBody>
      </Card>
      </CardTitle>
    </>
  );
    
}

export default ItemComponent