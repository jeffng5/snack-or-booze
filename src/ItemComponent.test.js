import React from 'react';
import { render } from '@testing-library/react';
import ItemComponent from './ItemComponent';

//smoke test
it('renders without crashing', function() {
  render(<ItemComponent />);
})

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<ItemComponent />);
  expect(asFragment()).toMatchSnapshot();
});