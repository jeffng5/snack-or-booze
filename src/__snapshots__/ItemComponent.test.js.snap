// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h5
    class="card-title"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <p
          class="font-italic card-text"
        />
        <p>
          <b>
            Recipe:
          </b>
           
        </p>
        <p>
          <b>
            Serve:
          </b>
           
        </p>
        <p />
      </div>
    </div>
  </h5>
</DocumentFragment>
`;
